evm-contracts/src/v0.7/dev/Operator.sol

```diff
@@ -373,19 +364,48 @@ contract Operator is
   )
   internal
   {
-    bytes32 paramsHash = keccak256(
+    bytes31 paramsHash = buildFunctionHash(payment, callbackAddress, callbackFunctionId, expiration);
+    require(s_commitments[requestId].paramsHash == paramsHash, "Params do not match request ID");
+    require(s_commitments[requestId].dataVersion <= safeCastToUint8(dataVersion), "Data versions must match");
+    s_withdrawableTokens = s_withdrawableTokens.add(payment);
+    delete s_commitments[requestId];
+  }
+
+  /**
+   * @notice Build the bytes31 function hash from the payment, callback and expiration.
+   * @param payment The payment amount that will be released for the oracle (specified in wei)
+   * @param callbackAddress The callback address to call for fulfillment
+   * @param callbackFunctionId The callback function ID to use for fulfillment
+   * @param expiration The expiration that the node should respond by before the requester can cancel
+   * @return hash bytes31
+   */
+  function buildFunctionHash(
+    uint256 payment,
+    address callbackAddress,
+    bytes4 callbackFunctionId,
+    uint256 expiration
+  )
+  internal
+  returns (bytes31)
+  {
+    return bytes31(keccak256(
       abi.encodePacked(
         payment,
         callbackAddress,
         callbackFunctionId,
         expiration
       )
-    );
-    require(s_commitments[requestId] == paramsHash, "Params do not match request ID");
-    require(s_dataVersions[requestId] <= dataVersion, "Incorrect data version");
-    s_withdrawableTokens = s_withdrawableTokens.add(payment);
-    delete s_commitments[requestId];
-    delete s_dataVersions[requestId];
+    ));
+  }
+
+  /**
+   * @notice Safely cast uint256 to uint8
+   * @param number uint256
+   * @return uint8 number
+   */
+  function safeCastToUint8(uint256 number) internal returns (uint8) {
+    require(number < 2**8, "number too big to cast");
```

Comment:
I'm not sure how the Solidity compiler will handle 2**8, if it will automatically make it a constant. I think it'd be worth pulling it out into a constant anyway just to give the number a name and make it more readable.